File: 0670-maximum-swap/0670-maximum-swap.java
class Solution {
    public int maximumSwap(int num) {
        
        int n = (int)Math.log10(num) + 1;
        int arr[] = new int[n];
        int maxArr[] = new int[n];
        int p = num, x = n - 1;
        maxArr[n-1] = n-1;
        while(p > 0)
        {
            arr[x--] = p % 10;
            p = p / 10;
        }
        int max = 0;
        for(int i = n-2; i >= 0; i--)
        {
            if(max < arr[i+1])
            {
                maxArr[i] = i+1;
                max = arr[i+1];
                continue;
            }
            maxArr[i] = maxArr[i+1];
        }
        for(int i = 0; i < n-1; i++)
        {
            if(arr[i] < arr[maxArr[i]])
            {
                int temp = arr[i];
                arr[i] = arr[maxArr[i]];
                arr[maxArr[i]] = temp;
                break;
            }
        }
        int ans = 0;
        for(int  i = 0; i < n; i++)
        {
            ans = ans * 10 +arr[i];
        }
        return ans;
    }
}

File: 0670-maximum-swap/0670-maximum-swap.cpp
class Solution {
public:
    int maximumSwap(int num) {
        
        int p = num;
        vector<int> arr;
        arr.push_back(0);
        string s = "";
        while(p != 0)
        {
            int r = p%10;
            s.push_back((char)(r+48));
            arr.push_back(max(r,arr[arr.size()-1]));
            p = p/10;
        }
        reverse(arr.begin(),arr.end());
        reverse(s.begin(),s.end());
        int pos1 = -1,v1 = -1, x = -1;
        for(int i = 0; i < s.length(); i++)
        {
            if(((int)(s[i]) - 48) < arr[i+1])
            {
                pos1 = i;
                v1 = arr[i+1];
                x = ((int)(s[i])-48);
                break;
            }
        }
        string w = "";
        bool f = false;
        int ans = 0;
        for(int i = s.length()-1; i >= 0; i--)
        {
            if(((int)(s[i])-48) == v1 && f==false)
            {
                w = (char)(x + 48) + w;
                f = true;
                continue;
            }
            if(i == pos1)
            {
                w = (char)(v1 + 48) + w;
                continue;
            }
            w = s[i] + w;
        }
        for(int i = 0; i < w.length(); i++)
        {
            ans =ans * 10 + (int)(w[i] - 48);   
        }
        return ans;
    }
};